freqtrade/freqtradebot.py

```diff
@@ -0,0 +1,535 @@
+"""
+Freqtrade is the main module of this bot. It contains the class Freqtrade()
+"""
+
+import copy
+import json
+import time
+import traceback
+from typing import Dict, List, Optional, Any, Callable
+from datetime import datetime
+import requests
+import arrow
+from cachetools import cached, TTLCache
+from freqtrade.analyze import Analyze
+from freqtrade.constants import Constants
+from freqtrade.fiat_convert import CryptoToFiatConverter
+from freqtrade.logger import Logger
+from freqtrade.persistence import Trade
+from freqtrade.rpc.rpc_manager import RPCManager
+from freqtrade.state import State
+from freqtrade import (DependencyException, OperationalException, exchange, persistence)
+
+
+class FreqtradeBot(object):
+    """
+    Freqtrade is the main class of the bot.
+    This is from here the bot start its logic.
+    """
+
+    def __init__(self, config: Dict[str, Any], db_url: Optional[str] = None) -> bool:
+        """
+        Init all variables and object the bot need to work
+        :param config: configuration dict, you can use the Configuration.get_config()
+        method to get the config dict.
+        :param db_url: database connector string for sqlalchemy (Optional)
+        """
+
+        # Init the logger
+        self.logger = Logger(name=__name__, level=config.get('loglevel')).get_logger()
+
+        # Init bot states
+        self._state = State.STOPPED
```

Comment:
I think we should mark `self._state` as public accessible, so we can drop `update_state()` and `get_state()`